docs/source/configuration/cors.mdx

```diff
@@ -1,13 +1,143 @@
+
+If two URLs differ in their domain, protocol, or port, then those URLs come from two different origins:
+
+```bash
+# Same origin
+http://example.com:8080/ <==> http://example.com:8080/
+
+# Different origin (difference in domain, protocol, and port)
+http://example.com:8080/ =X= https://example1.com:8081/
+```
+
+However, as we all know, the internet is an exciting place full of resources that can make websites better (importing images, extra fonts, making API calls, and so on).  Developers needed a new protocol to relax SOP and safely _share_ resources across different origins.
+
+Cross-Origin Resource Sharing is the mechanism that allows a web page to share resources across different origins. CORS provides an extra layer of protection by enabling servers and clients to define HTTP headers that specify _which_ external clients' scripts can access their resources.
+
+Note that both SOP and CORS are related to [_browser_ security](https://developer.mozilla.org/en-US/docs/Web/Security#security-related_glossary_terms). Neither prevents _other_ types of software from requesting resources from your server.
+
+## Choosing CORS options for your project
+
+When thinking about configuring CORS for your application, there are two main settings to consider:
+
+* Which origins can access your server's resources
+* Whether your server accepts user credentials (i.e., cookies) with requests
+
+### Specifying origins
+
+CORS uses [specific HTTP request headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers) as part of its protocol, including [`Access-Control-Allow-Origin`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin). The `Access-Control-Allow-Origin` header (ACAO) enables a server to dictate which origins can use scripts to access that server's resources.
+
+Depending on what you're building, the origins you specify in your CORS configuration might need to change when you're ready to deploy your application.
+
+#### Applications on private networks
+
+If your browser is running your API on a private network (i.e., not on the public internet) and it relies on the privacy of that network for security, we strongly recommend [specifying which origins](#configuring-cors-options-for-apollo-server) can access your server's resources.
+
+If you instead rely on `allow_any_origin`, while your personal computer is on your private network, a script on any website could potentially make your browser talk to your private API. [Some browsers, such as Chrome, have features under development](https://wicg.github.io/private-network-access/) to solve this problem. But in the meantime, all servers on private networks should *always* specify origins in their CORS configuration.
```

Comment:
Indent with `>` as this paragraph feels like a warning.

Reply:
fixed in 71bf26c8